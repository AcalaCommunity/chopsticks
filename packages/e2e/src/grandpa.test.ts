import { describe, expect, it } from 'vitest'

import { api, delay, dev, mockCallback, setupApi } from './helper.js'

setupApi({
  endpoint: ['wss://rpc.ibp.network/polkadot'],
  blockHash: '0xb012d04c56b65cfa1f47cb1f884d920f95d0097b1ed42f5da18d5e2a436c2f3e',
})

describe('grandpa rpc', () => {
  it('subscribeJustifications', async () => {
    const { callback, next } = mockCallback()
    const unsub = await api.rpc.grandpa.subscribeJustifications(callback)

    await next()
    expect(callback.mock.calls).toMatchSnapshot()

    callback.mockClear()

    expect(await dev.newBlock()).toMatchSnapshot()

    await next()

    expect(callback.mock.calls).toMatchSnapshot()

    callback.mockClear()

    unsub()

    expect(await dev.newBlock()).toMatchSnapshot()

    await delay(100)

    expect(callback).not.toHaveBeenCalled()
  })
})
