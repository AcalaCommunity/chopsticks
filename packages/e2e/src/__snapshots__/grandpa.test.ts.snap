// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`grandpa rpc > subscribeJustifications 1`] = `
[
  [
    "0x7777777777777777777777777777777777777777777777777777777777777777",
  ],
]
`;

exports[`grandpa rpc > subscribeJustifications 2`] = `"0x7ba5ba52f5b88fa5bcab5c1a0f7e158d845d93e1c70e9a5b42a7c3a9af681ee7"`;
